src/core/Akka.Persistence/Eventsourced.Recovery.cs

```diff
@@ -17,17 +17,15 @@ namespace Akka.Persistence
 
     internal class EventsourcedState
     {
-        public EventsourcedState(string name, bool isRecoveryRunning, StateReceive stateReceive)
+        public EventsourcedState(string name, Func<bool> isRecoveryRunning, StateReceive stateReceive)
         {
             Name = name;
             IsRecoveryRunning = isRecoveryRunning;
             StateReceive = stateReceive;
         }
 
         public string Name { get; }
-
-        public bool IsRecoveryRunning { get; }
-
+        public Func<bool> IsRecoveryRunning { get; }
```

Comment:
This breaks the public API

Reply:
Nah, this is an `internal` API. None of the third party Akka.Persistence plugins have access to this so we're able to preserve binary compatibility here. It's only the Akka.Persistence engine itself that consumes this directly so I think you're fine.